scipy/stats/tests/test_axis_nan_policy.py

```diff
@@ -695,6 +696,59 @@ def test_masked_array_2_sentinel_array():
     assert B_out is B
 
 
+def test_masked_dtype():
```

Comment:
I know this is lengthy, but I thought it would be more readable and require less test execution time if I wrote very specific tests rather than writing shorter but more complex code to test everything under the sun. Hopefully the comments help with interpretation.